contracts/src/provers/HashiProver.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {RLPReader} from "optimism/packages/contracts-bedrock/src/libraries/rlp/RLPReader.sol";
+import {IProver} from "../interfaces/IProver.sol";
+import {IShoyuBashi} from "../interfaces/IShoyuBashi.sol";
+import {StateValidator} from "../libraries/StateValidator.sol";
+import {RIP7755Inbox} from "../RIP7755Inbox.sol";
+import {CrossChainRequest} from "../RIP7755Structs.sol";
+
+/// @title HashiProver
+///
+/// @author Crosschain Alliance
+///
+/// @notice This contract implements storage proof validation to ensure that requested calls actually happened on a chain where Hashi is available
+contract HashiProver is IProver {
+    using StateValidator for address;
+    using RLPReader for RLPReader.RLPItem;
+    using RLPReader for bytes;
+
+    /// @notice Parameters needed for a full nested cross-chain storage proof
+    struct RIP7755Proof {
+        /// @dev The parameter specifies the chain ID for retrieving the block header from Hashi
+        uint256 dstChainId;
```

Comment:
We don't need `dstChainId` here as it's already in the request struct being passed in